agent/src/main/java/com/appland/appmap/process/hooks/Spark.java

```diff
@@ -145,11 +145,16 @@ public static void setHandler(Event event, Object receiver, Object[] args) {
     if (!argClass.equals(jettyHandlerClass)) {
       if (argClass.equals("org.eclipse.jetty.server.handler.HandlerList")) {
         HandlerList hl = new HandlerList(handler);
-        boolean match = Arrays.stream(hl.getHandlers()).anyMatch(h -> h.getClass().getName().equals(jettyHandlerClass));
+        boolean match = Arrays.stream(hl.getHandlers())
+            .anyMatch(h -> h.getClass().getName().equals(jettyHandlerClass));
         if (!match) {
           return;
         }
       }
+
+      // If it's not either the Spark JettyHandler or a HandlerList, we're not running in a Spark
+      // server.
+      return;
```

Comment:
Won't this always return for a HandlerList regardless whether it contains a JettyHandler? 
Perhaps this logic would be clearer in a separate utility function.